app/scripts/lib/account-tracker.js

```diff
@@ -158,90 +280,148 @@ export default class AccountTracker {
+    Object.keys(accountsByChainId).forEach((chainId) => {
+      accountsByChainId[chainId] = {};
+    });
+    this.store.updateState({ accounts: {}, accountsByChainId });
   }
 
   /**
-   * Given a block, updates this AccountTracker's currentBlockGasLimit, and then updates each local account's balance
-   * via EthQuery
+   * Given a block, updates this AccountTracker's currentBlockGasLimit and currentBlockGasLimitByChainId and then updates
+   * each local account's balance via EthQuery
    *
    * @private
    * @param {number} blockNumber - the block number to update to.
    * @fires 'block' The updated state, if all account updates are successful
    */
   async _updateForBlock(blockNumber) {
-    this._currentBlockNumber = blockNumber;
+    await this._updateForBlockByNetworkClientId(null, blockNumber);
+  }
+
+  /**
+   * Given a block, updates this AccountTracker's currentBlockGasLimitByChainId, and then updates each local account's balance
+   * via EthQuery
+   *
+   * @private
+   * @param {string} networkClientId - optional network client ID to use instead of the globally selected network.
+   * @param {number} blockNumber - the block number to update to.
+   * @fires 'block' The updated state, if all account updates are successful
+   */
+  async _updateForBlockByNetworkClientId(networkClientId, blockNumber) {
+    const { chainId, provider } =
+      this.#getCorrectNetworkClient(networkClientId);
+    this._currentBlockNumberByChainId[chainId] = blockNumber;
 
     // block gasLimit polling shouldn't be in account-tracker shouldn't be here...
-    const currentBlock = await this._query.getBlockByNumber(blockNumber, false);
+    const currentBlock = await pify(new EthQuery(provider)).getBlockByNumber(
+      blockNumber,
+      false,
+    );
     if (!currentBlock) {
       return;
     }
     const currentBlockGasLimit = currentBlock.gasLimit;
-    this.store.updateState({ currentBlockGasLimit });
+    const { currentBlockGasLimitByChainId } = this.store.getState();
+    currentBlockGasLimitByChainId[chainId] = currentBlockGasLimit;
+    this.store.updateState({ currentBlockGasLimitByChainId });
+    if (chainId === this.getCurrentChainId()) {
+      this.store.updateState({
+        currentBlockGasLimit,
+      });
+    }
 
     try {
-      await this._updateAccounts();
+      await this._updateAccounts(networkClientId);
     } catch (err) {
       log.error(err);
     }
   }
 
+  async _updateAccountsAllActiveNetworks() {
+    this._updateAccounts();
+    this.#pollingTokenSets.forEach((_tokenSet, networkClientId) => {
+      this._updateAccounts(networkClientId);
+    });
+  }
+
   /**
    * balanceChecker is deployed on main eth (test)nets and requires a single call
    * for all other networks, calls this._updateAccount for each account in this.store
    *
+   * @param {string} networkClientId - optional network client ID to use instead of the globally selected network.
```

Comment:
[nit]
```suggestion
   * @private
   * @param {string} networkClientId - optional network client ID to use instead of the globally selected network.
```

Reply:
Not private. Called in MetaMaskController via networkDidChange

https://github.com/MetaMask/metamask-extension/blob/10996f0aaaf3a1d469cd79942c789e07152ad8f7/app/scripts/metamask-controller.js#L1646-L1652

Reply:
oh lol. I don't think it should have the underscore prefix then...

Reply:
yup 🙃 .

I'll change it